lib/hyacinth/utils/csv_import_export_utils.rb

```diff
@@ -175,6 +183,117 @@ def self.create_import_job_from_csv_data(csv_data_string, import_filename, user)
     import_job
   end
 
+  # Takes a csv data string and returns a map that, for each row, lists
+  # prerequisite rows
+  def self.prerequisite_row_map_for_csv_data(csv_data_string)
+    index_of_parent_headers = []
+    index_of_pid_header = nil
+
+    row_number_for_header_row = 0
+
+    # Get index of parent headers if present
+    CSV.parse(csv_data_string) do |row|
+      unless row[0].start_with?('_')
+        # Go to second row if first row doesn't contain headers
+        row_number_for_header_row += 1
+        next
+      end
+      row.each_with_index do |header, index|
+        index_of_parent_headers << index if header =~ /^_parent_digital_objects-\d+\.(identifier|pid)$/
+        index_of_pid_header = index if header == '_pid'
+      end
+      break
+    end
+
+    return {} if index_of_parent_headers.blank?
+
+    identifiers_to_row_numbers = identifiers_to_row_numbers_for_csv_data(csv_data_string)
+
+    # Return if no identifiers were found in the csv data, since that means that processing order doesn't matter
+    return {} if identifiers_to_row_numbers.blank?
+
+    prerequisite_rows_based_on_headers(csv_data_string, row_number_for_header_row, index_of_parent_headers, identifiers_to_row_numbers)
+  end
+
+  def self.prerequisite_rows_based_on_headers(csv_data_string, row_number_for_header_row, index_of_parent_headers, identifiers_to_row_numbers)
```

Comment:
This method is starting to raise warning flags because of the all parameters required, but I totally see why it's needed. I almost wonder if a intermediate data structure would be helpful to make the code easier to read.

Reply:
I'm 👍  @cgalarza on this - the class is getting a little bulky in general. Maybe out of scope for this PR, but I'd definitely be in favor of whittling it down a little.

Reply:
This many-param method comes from me having to refactor around rubocop's Cyclomatic Complexity rule.  Too many conditionals/branches in a single method.  I wish the variables were just in an outer scope and didn't need to be passed.  I'm also in favor of whittling down later on.